projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts

```diff
@@ -1762,5 +1762,19 @@ describe('IgxTreeGrid - Integration #tGrid', () => {
             firstRow.pinned = true;
             expect(firstRow.pinned).toBeTrue();
         });
+
+        it('should delete pinned row without errors', () => {
+            treeGrid.pinRow(147);
+            fix.detectChanges();
+            const firstRow = treeGrid.rowList.toArray()[0];
+
+            expect(firstRow.isRoot).toBe(true);
+            expect(firstRow.pinned).toBeTrue();
```

Comment:
Not sure if this is a good test - if there are ANY pinned records, treeGrid.rowList.toArray()[0].pinned will always return true. In adittion to that, verify that this pinned row has an ID of 147.